app/plugin.go

```diff
@@ -632,3 +637,63 @@ func (a *App) notifyPluginEnabled(manifest *model.Manifest) error {
 
 	return nil
 }
+
+func (a *App) signaturesFromPathToReader(plugin *pluginSignaturePaths) (signatures []io.ReadSeeker, closeFunc func(), appErr *model.AppError) {
+	signatures = []io.ReadSeeker{}
+	for _, signaturePath := range plugin.signaturePaths {
+		sigReader, appErr := a.FileReader(signaturePath)
+		if appErr != nil {
+			mlog.Debug("Can't read signature file.", mlog.String("path", signaturePath), mlog.Err(appErr))
+			return nil, nil, appErr
+		}
+		signatures = append(signatures, sigReader)
+	}
+
+	closeFunc = func() {
+		for _, signature := range signatures {
+			readCloseSeeker, ok := signature.(filesstore.ReadCloseSeeker)
+			if !ok {
+				mlog.Debug("Failed to cast signature to ReadCloseSeeker.", mlog.String("paths", fmt.Sprint(plugin.signaturePaths)), mlog.String("pluginId", plugin.pluginId))
+			}
+			readCloseSeeker.Close()
+		}
+	}
+	return signatures, closeFunc, nil
+}
+
+func (a *App) getPluginsFromFolder() (map[string]*pluginSignaturePaths, *model.AppError) {
+	fileStorePaths, appErr := a.ListDirectory(fileStorePluginFolder)
+	if appErr != nil {
+		return nil, model.NewAppError("getPluginsFromDir", "app.plugin.sync.list_filestore.app_error", nil, appErr.Error(), http.StatusInternalServerError)
+	}
+	if len(fileStorePaths) == 0 {
+		mlog.Info("Found no files in plugins file store")
```

Comment:
nit: This log only gets emitted when we find no plugins, but we don't emit one (from this function) when we do find plugins. Should we just omit the log altogether? Should we log at a debug level more verbosely when we do find plugins below?

Reply:
The logging was in the previous implementation, so I did not change. I'll add the found plugins log, but should I remove this one?

Reply:
I'd be fine with removing this log and skipping adding any new "found" log, since the side effects should be obvious and logged anyway.